src/sage/combinat/integer_vector.py

```diff
@@ -1043,11 +1043,21 @@ def __iter__(self):
              [2, 1],
              [1, 2],
              [0, 3]]
+
+        TESTS:
+
+        Check corner case::
+
+            sage: IV = IntegerVectors(k=0)
+            sage: list(IV)
+            [[]]
         """
         n = 0
         while True:
             for iv in integer_vectors_nk_fast_iter(n, self.k):
                 yield self.element_class(self, iv, check=False)
+            if self.k == 0:
+                return
```

Comment:
```suggestion
        n = 0
        if self.k == 0:  # special case
                yield self.element_class(self, [], check=False)
                return
        while True:
            for iv in integer_vectors_nk_fast_iter(n, self.k):
                yield self.element_class(self, iv, check=False)
```
I think this makes it a bit more clear about the special case.

Reply:
Probably also a good idea, then if k ≠ 0 it isn't necessary to check the condition every iteration.